src/test/java/restfulbooking/BookingTest.java

```diff
@@ -0,0 +1,207 @@
+package restfulbooking;
+
+import kong.unirest.Headers;
+import kong.unirest.HttpResponse;
+import kong.unirest.JsonNode;
+import kong.unirest.Unirest;
+import kong.unirest.json.JSONArray;
+import kong.unirest.json.JSONObject;
+import org.junit.Test;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+public class BookingTest
+{
+//	String restfulBookerEndpointBaseURL = "https://restful-booker.herokuapp.com";
+	String restfulBookerEndpointBaseURL = "http://localhost:3001";
+	String authEndpoint =  restfulBookerEndpointBaseURL + "/auth";
+	String bookingEndpoint = restfulBookerEndpointBaseURL + "/booking";
+
+	@Test
+	public void shouldReturnListOfBookings()
+	{
+		JSONArray bookingsList = getBookingsList();
+
+		// assertions are in getBookingsList() helper method
+	}
+
+	public JSONArray getBookingsList()
+	{
+		HttpResponse<JsonNode> response = Unirest.get(bookingEndpoint)
+				.header("Accept", "application/json")
+				.asJson();
+
+		int status = response.getStatus();
+		Headers headers = response.getHeaders();
+		JsonNode body = response.getBody();
+
+//		System.out.println(status);
+//		System.out.println(headers);
+//		System.out.println(body);
+
+		assertThat(status).isEqualTo(200);
+		assertThat(headers.getFirst("Content-type")).startsWith("application/json");
+		assertThat(body).isNotNull();
+
+		assertThat(body.isArray());
+
+		JSONArray bookings = body.getArray();
+		System.out.println("number of bookings: " + bookings.length());
+
+		assertThat(bookings.length()).isNotNegative();
+
+		return bookings;
+	}
+
+	@Test
+	public void shouldHaveBookingDetailsForEachBooking()
```

Comment:
@sauceaaron so this is the test that would be your preferred way of writing tests with an API in Java?